RecoEgamma/EgammaPhotonAlgos/src/ConversionTrackEcalImpactPoint.cc

```diff
@@ -116,15 +116,14 @@ std::vector<math::XYZPointF> ConversionTrackEcalImpactPoint::find( const std::ve
       int ibc=0;
       goodBC=0;
 
-      for (unsigned i = 0; i < bcHandle->size(); ++i ) {
-	float dEta= bcHandle->ptrAt(i)->position().eta() - ecalImpactPosition.eta()  ;
-	float dPhi= bcHandle->ptrAt(i)->position().phi() - ecalImpactPosition.phi()  ;
-	if ( sqrt(dEta*dEta + dPhi*dPhi)  <  bcDistanceToTrack ) {
+      for(auto const& bc : bcHandle->ptrs()) {
+        float dEta= bc->position().eta() - ecalImpactPosition.eta();
+        float dPhi= bc->position().phi() - ecalImpactPosition.phi();
```

Comment:
The function we are talking about here by the way... I think it might have a bug. The track iterator gets incremented in the for loop declaration [1] and at the end of the for loop [2].

Got no time to check that out now, but I think we should scrutinize this later.

[1] https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaPhotonAlgos/src/ConversionTrackEcalImpactPoint.cc#L83
[2] https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaPhotonAlgos/src/ConversionTrackEcalImpactPoint.cc#L134

Reply:
Sorry, looking at it again I realized that I was just not reading well and confused the variable `iTk` with `iTrk`...